dune/stuff/functions/expression.hh

```diff
@@ -303,6 +340,21 @@ private:
       }
   } // ... build_gradients(...)
 
+  template< int rows, int cols >
+  void build_gradients(const std::string variable,
+                       Dune::FieldMatrix< std::string, rows, cols > jacobian)
+  {
+    assert(jacobian.rows == 0 || jacobian.rows >= dimRange);
```

Comment:
Actually, the check is needed because the method is templated by rows and cols, I think.